### PR TITLE
RateLimiter improvements

### PR DESCRIPTION
**Pull request by kevingosse**
- Uses UTC time to prevent issues during DST transitions
- Removes locking

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace/Sampling/RateLimiter.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public RateLimiter(int? maxTracesPerInterval)
             _maxTracesPerInterval = maxTracesPerInterval ?? 100;
             _intervalMilliseconds = 1_000;
             _interval = TimeSpan.FromMilliseconds(_intervalMilliseconds);
-            _windowBegin = _lastRefresh = DateTime.Now;
+            _windowBegin = DateTime.UtcNow;
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Sampling/RateLimiter.cs:33]: Good catch!

File: src/Datadog.Trace/Sampling/RateLimiter.cs [hunk 2]
```diff
@@ -143,8 +137,13 @@ private void WaitForRefresh()
...
             finally
             {
-                // Resume threads
-                _refreshEvent.Set();
+                if (refreshing == 0)
+                {
+                    // If refreshing is 0, it means that this thread acquired the lock
+                    // Releasing it before leaving
+                    // Note: a full fence might not be needed here, but better safe than sorry
+                    Interlocked.Exchange(ref _refreshing, 0);
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Sampling/RateLimiter.cs:145]: When I first read this I though we were setting the value to zero if it is already zero, which made no sense.  Then I realized `refreshing` and `_refreshing` are two different things. Can we rename them to something a little more clear? Maybe the local variable can be something like `wasRefreshing` or `lockAquired`?

Comment (kevingosse) [reply src/Datadog.Trace/Sampling/RateLimiter.cs:145]: Does `refreshInProgress` work?

**Comment by zacharycmontoya** [comment]
I initiated a re-run of the unit-tests pipeline since my latest PR should fix that failing test.

**Pull request merged**